lib/bob_dns.ml

```diff
@@ -51,38 +54,60 @@ module Transport :
         in
         let socket = Unix.socket fam Unix.SOCK_STREAM p_proto in
         let addr = Unix.ADDR_INET (Ipaddr_unix.to_inet_addr ip, port) in
-        Fiber.connect socket addr >>= function
-        | Error _errno ->
-            Fiber.return
-              (Some (Happy_eyeballs.Connection_failed (host, id, (ip, port))))
-        | Ok () -> (
-            let waiters, r =
+        Fiber.pick
+          (fun () ->
+            Fiber.connect socket addr >>= function
```

Comment:
Since `Fiber.pick` doesn't cancel, could there be issues from `Fiber.connect` still pending when the socket is closed?

Reply:
I would like to believe that if we call `Unix.close` on the socket, `Unix.connect` will signals an error from `Unix.select`. In that case, we will probably resolve `Fiber,connect` with this error. It's not totally clear that the situation will be scheduled like this mainly because the `Unix.connect`'s behavior depends on the system but I would like to hope on that.